<?xml version="1.0"?>
<robot  xmlns:xacro="http://www.ros.org/wiki/xacro" name="firefighter" >
	
<xacro:property name="width" value=".2" />

  <link name="robot_base">
      <inertial>
      <mass value="0.00"/>
      <inertia ixx="0.00002" ixy="0" ixz="0" iyy="0.00002" iyz="0" izz="0.00002"/>
      <origin xyz = "0.0 0.0 0.00" rpy = "0 0 0"/>
    </inertial>
  </link>

  <link name="g_base">
    <visual>
      <geometry>
	     <cylinder length="0.029" radius="0.05"/>
      </geometry>
      <origin xyz = "0.0 0 0.0145" rpy = " 0 0 0"/>
    </visual>
    <collision>
      <geometry>
	     <cylinder length="0.029" radius="0.05"/>
      </geometry>
      <origin xyz = "0.0 0 0.0145" rpy = " 0 0 0"/>
    </collision>
    <inertial>
      <mass value="0.025"/>
      <inertia ixx="0.00002" ixy="0" ixz="0" iyy="0.00002" iyz="0" izz="0.00002"/>
      <origin xyz = "0.0 0 0.0145" rpy = " 0 0 0"/>
    </inertial>
  </link>

  <link name="joint1">
    <visual>
      <geometry>
	     <!--- 0.0 0 -0.04  1.5708 3.14159-->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes/joint1.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 0 0 0"/>
    </visual>
    <collision>
      <geometry>
	     <!--- 0.0 0 -0.04  1.5708 3.14159-->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint1.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0 " rpy = " 0 0 0"/>
    </collision>
    <inertial>
      <mass value="0.2"/>
      <inertia ixx="0.00005" ixy="0" ixz="0" iyy="0.00005" iyz="0" izz="0.00005"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
    </inertial>
  </link>

  <link name="joint2">
    <visual>
      <geometry>
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint2.dae"/>
      </geometry>
    <origin xyz = "0.0 0 -0.06096 " rpy = " 0 0 -1.5708"/>
    </visual>
      <collision>
      <geometry>
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint2.dae"/>
      </geometry>
    <origin xyz = "0.0 0 -0.06096 " rpy = " 0 0 -1.5708"/>
    </collision>
    <inertial>
      <mass value="0.2"/>
      <inertia ixx="0.00005" ixy="0" ixz="0" iyy="0.00005" iyz="0" izz="0.00005"/>
      <origin xyz="0 0 -0.03048" rpy="0 0 0"/>
    </inertial>
  </link>

  <link name="joint3">
    <visual>
      <geometry>
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint3.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0.03256 " rpy = " 0 -1.5708 0"/>
    </visual>
      <collision>
        <geometry>
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint3.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0.03256 " rpy = " 0 -1.5708 0"/>
    </collision>
     <inertial>
      <mass value="0.2"/>
      <inertia ixx="0.00005" ixy="0" ixz="0" iyy="0.00005" iyz="0" izz="0.00005"/>
      <origin xyz="0 0 0.01628" rpy="0 0 0"/>
    </inertial>
  </link>

  <link name="joint4">
    <visual>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint4.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0.03056 " rpy = " 0 -1.5708 0"/>
    </visual>
      <collision>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint4.dae"/>
      </geometry>
    <origin xyz = "0.0 0 0.03056 " rpy = " 0 -1.5708 0"/>
    </collision>
     <inertial>
      <mass value="0.2"/>
      <inertia ixx="0.00005" ixy="0" ixz="0" iyy="0.00005" iyz="0" izz="0.00005"/>
      <origin xyz="0 0 0.01528" rpy="0 0 0"/>
    </inertial>
  </link>

  <link name="joint5">
    <visual>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint5_new.dae"/>
      </geometry>
    <origin xyz = "0.0 0 -0.03356 " rpy = " 0 -1.5708 1.5708"/>
    </visual>
      <collision>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint5_new.dae"/>
      </geometry>
    <origin xyz = "0.0 0 -0.03356 " rpy = " 0 -1.5708 1.5708"/>
    </collision>
    <inertial>
      <mass value="0.2"/>
      <inertia ixx="0.00005" ixy="0" ixz="0" iyy="0.00005" iyz="0" izz="0.00005"/>
      <origin xyz="0 0 -0.01678" rpy="0 0 0"/>
    </inertial>
  </link>

  <link name="joint6">
    <visual>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint6.dae"/>
      </geometry>
    <origin xyz = "0 0.00 -0.038 " rpy = " 0 0 0"/>
    </visual>
      <collision>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint6.dae"/>
      </geometry>
    <origin xyz = "0 0.00 -0.038 " rpy = " 0 0 0"/>
    </collision>
     <inertial>
      <mass value="0.2"/>
      <inertia ixx="0.00005" ixy="0" ixz="0" iyy="0.00005" iyz="0" izz="0.00005"/>
      <origin xyz="0 0 -0.019" rpy="0 0 0"/>
    </inertial>
  </link>

  <link name="joint6_flange">
    <visual>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint7.dae"/>
      </geometry>
    <origin xyz = "0.0 0 -0.012 " rpy = " 0 0 0"/>
    </visual>
      <collision>
      <geometry>
       <!--- 0.0 0 -0.04 -->
       <mesh filename="package://ros_sim/resource/robots/mycobot_280_pi/meshes//joint7.dae"/>
      </geometry>
    <origin xyz = "0.0 0 -0.012 " rpy = " 0 0 0"/>
    </collision>
    <inertial>
      <mass value="0.05"/>
      <inertia ixx="0.00002" ixy="0" ixz="0" iyy="0.00002" iyz="0" izz="0.00002"/>
      <origin xyz="0 0 -0.006" rpy="0 0 0"/>
    </inertial>
  </link>

  <link name="wand">
    <visual>
      <geometry>
	     <cylinder length="0.14" radius="0.005"/>
      </geometry>
      <origin xyz = "0.0 0 -0.07" rpy = "0 0 0"/>
      <material name="red">
        <color rgba= "1 0 0 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
	     <cylinder length="0.14" radius="0.005"/>
      </geometry>
      <origin xyz = "0.0 0.0 -0.07" rpy = "0 0 0"/>
    </collision>

    <inertial>
      <mass value="0.025"/>
      <inertia ixx="0.00002" ixy="0" ixz="0" iyy="0.00002" iyz="0" izz="0.00002"/>
      <origin xyz = "0.0 0.0 -0.07" rpy = "0 0 0"/>
    </inertial>
  </link>

  <link name="wand_tip">
    <inertial>
      <mass value="0.00"/>
      <inertia ixx="0.00002" ixy="0" ixz="0" iyy="0.00002" iyz="0" izz="0.00002"/>
      <origin xyz = "0.0 0.0 0.00" rpy = "0 0 0"/>
    </inertial>
  </link>


  <joint name="robot_base_to_g_base" type="fixed">
    <parent link="robot_base"/>
    <child link="g_base"/>
    <origin xyz= "0 0 0.003" rpy = "0 0 0"/>  
  </joint>

  <joint name="g_base_to_joint1" type="fixed">
    <axis xyz="0 0 0"/>
    <limit effort = "1000.0" lower = "-3.14" upper = "3.14159" velocity = "0"/>
    <parent link="g_base"/>
    <child link="joint1"/>
    <origin xyz= "0 0 0.026" rpy = "0 0 0"/>  
  </joint>

  <joint name="joint2_to_joint1" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort = "1000.0" lower = "-2.87925" upper = "2.87925" velocity = "2.792527"/>
    <parent link="joint1"/>
    <child link="joint2"/>
    <origin xyz= "0 0 0.13956" rpy = "0 0 0"/>  
    <dynamics damping="0.2"/>
    <command_interface name="position"/>
    <state_interface name="position">
      <param name="initial_value">0.0</param>
    </state_interface>

  </joint>

  <joint name="joint3_to_joint2" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort = "1000.0" lower = "-2.87925" upper = "2.87925" velocity = "2.792527"/>
    <parent link="joint2"/>
    <child link="joint3"/>
    <origin xyz= "0 0 -0.001" rpy = "0 1.5708 -1.5708"/>  
    <dynamics damping="0.2"/>

    <command_interface name="position"/>
    <state_interface name="position">
      <param name="initial_value">0.0</param>
    </state_interface>

  </joint>

  <joint name="joint4_to_joint3" type="revolute">
    <axis xyz=" 0 0 1"/>
    <limit effort = "1000.0" lower = "-2.87925" upper = "2.87925" velocity = "2.792527"/>
    <parent link="joint3"/>
    <child link="joint4"/>
    <origin xyz= "  -0.1104 0 0   " rpy = "0 0 0"/> 
    <dynamics damping="0.2"/>

    <command_interface name="position"/>
    <state_interface name="position">
      <param name="initial_value">0.0</param>
    </state_interface>

  </joint>

  <joint name="joint5_to_joint4" type="revolute">
    <axis xyz=" 0 0 1"/>
    <limit effort = "1000.0" lower = "-2.87925" upper = "2.87925" velocity = "2.792527"/>
    <parent link="joint4"/>
    <child link="joint5"/>
    <origin xyz= "-0.096 0 0.06462" rpy = "0 0 -1.5708"/>  
    <dynamics damping="0.2"/>

    <command_interface name="position"/>
    <state_interface name="position">
      <param name="initial_value">0.0</param>
    </state_interface>

  </joint>

  <joint name="joint6_to_joint5" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort = "1000.0" lower = "-2.87925" upper = "2.87925" velocity = "2.792527"/>
    <parent link="joint5"/>
    <child link="joint6"/>
    <origin xyz= "0 -0.07318 0" rpy = "1.5708 -1.5708 0"/>  
    <dynamics damping="0.2"/>
      <command_interface name="position"/>
    <state_interface name="position">
     <param name="initial_value">0.0</param>
    </state_interface>
  </joint>

  <joint name="joint6output_to_joint6" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort = "1000.0" lower = "-3.05375" upper = "3.05375" velocity = "2.792527"/>
    <parent link="joint6"/>
    <child link="joint6_flange"/>
    <origin xyz= "0 0.0456 0" rpy = "-1.5708 0 0"/>  
    <dynamics damping="0.2"/>
    <command_interface name="position"/>
    <state_interface name="position">
      <param name="initial_value">0.0</param>
    </state_interface>

  </joint>

  <!--- note how the pen is rotated versus wrist when set to 0.0 with real robot -->
  <joint name="joint6output_to_wand" type="fixed">
    <parent link="joint6_flange"/>
    <child link="wand"/>
    <origin xyz= "-0.028284271 -0.028284271 0.035" rpy = "0.0 0.0 0.0"/>  
  </joint>

  <joint name="wand_to_wand_tip" type="fixed">
    <parent link="wand"/>
    <child link="wand_tip"/>
    <origin xyz= "0.0 0.0 0.0" rpy = "0.0 0.0 0.0"/>  
  </joint>

</robot>
